server/src/http/instance_data.rs

```diff
@@ -0,0 +1,122 @@
+use chrono::Duration;
```

Comment:
Moving this to a new file in this PR makes this really challenging to review. I'm guessing some stuff has changed here?